src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -501,47 +501,83 @@ fx_ver_t fx_muxer_t::resolve_framework_version(const std::vector<fx_ver_t>& vers
     const pal::string_t& fx_ver,
     const fx_ver_t& specified,
     const bool& patch_roll_fwd,
-    const int& roll_fwd_on_no_candidate_fx)
+    const enum roll_fwd_on_no_candidate_fx_option& roll_fwd_on_no_candidate_fx)
 {
     trace::verbose(_X("Attempting FX roll forward starting from [%s]"), fx_ver.c_str());
 
     fx_ver_t most_compatible = specified;
     if (!specified.is_prerelease())
```

Comment:
>!specified.is_prerelease() [](start = 8, length = 26)

On [https://github.com/dotnet/core-setup/issues/3469](https://github.com/dotnet/core-setup/issues/3469), you said "When a preview build is specified we only roll forward if the specified preview version doesn't exist" but here it seems like all of the roll forward logic is disabled when a preview is specified. What am I missing?

Reply:
I'm missing the `else` of course. :)

---
In reply to: [157608241](https://github.com/dotnet/core-setup/pull/3537#discussion_r157608241) [](ancestors = 157608241)